backend/endpoints/rom.py

```diff
@@ -73,6 +83,21 @@ def rom(request: Request, id: int) -> RomSchema:
     return dbh.get_rom(id)
 
 
+@protected_route(router.put, "/platforms/{p_slug}/roms/upload", ["roms.write"])
+def upload_roms(request: Request, p_slug: str, roms: list[UploadFile] = File(...)):
+    platform_fs_slug = dbh.get_platform(p_slug).fs_slug
+    log.info(f"Uploading files to: {platform_fs_slug}")
+    if roms is not None:
+        roms_path = build_upload_roms_path(platform_fs_slug)
+        for rom in roms:
+            log.info(f" - Uploading {rom.filename}")
+            file_location = f"{roms_path}/{rom.filename}"
+            with open(file_location, "wb+") as file_object:
+                file_object.write(rom.file.read())
```

Comment:
What happens if there's a name collision? This would overwrite the contents of the file right?

Also how does this handle uploading large files? Is the entire content read into memory before being written?

Reply:
Good ones, need to check

Reply:
> What happens if there's a name collision? This would overwrite the contents of the file right?

Now when uploading files the enpoint checks and raises an exception if any exists
 
> Also how does this handle uploading large files? Is the entire content read into memory before being written?
Now the files are read by chunks of 1024bytes